### PR TITLE
Enabling log file rotation

### PR DESCRIPTION
**Pull request by 09harsh**
# Description
implemented log rotation as part of this PR. Is solved the issue #664

## Type of change

*Please select one of the options below.*

- Bug fix (non-breaking change which fixes an issue)
[x]New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation
- Other

# Discussion

*Are there any design details that you would like to discuss further?*
NO

# Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*

# Documentation

*Did you update relevant documentation within this repository?*

*If this change is adding new functionality, do we need to describe it in our user manual?*

*If this change is adding or removing subreconcilers, have we updated the core technical design doc to reflect that?*

*If this change is adding new safety checks or new potential failure modes, have we documented and how to debug potential issues?*

# Follow-up

*Are there any follow-up issues that we should pursue in the future?*

*Does this introduce new defaults that we should re-evaluate in the future?*

### CODE REVIEW

File: setup/setup.go [hunk 1]
```diff
@@ -89,13 +98,14 @@ func StartManager(
...
 
 	if operatorOpts.LogFile != "" {
-		var err error
-		file, err = os.OpenFile(operatorOpts.LogFile, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0600)
-		if err != nil {
-			_, _ = os.Stderr.WriteString(err.Error())
-			os.Exit(1)
+		lumberjackLogger := &lumberjack.Logger{
+			Filename:   operatorOpts.LogFile,
+			MaxSize:    operatorOpts.LogFileMaxAge,
```
Comment (johscheuer) [inline setup/setup.go:103]:
```suggestion
			MaxSize:    operatorOpts.LogFileMaxSize,
```

**Review by johscheuer — CHANGES_REQUESTED** [review]
One small thing

File: setup/setup.go [hunk 2]
```diff
@@ -72,6 +77,10 @@ func (o *Options) BindFlags(fs *flag.FlagSet) {
 	fs.IntVar(&o.MaxConcurrentReconciles, "max-concurrent-reconciles", 1, "Defines the maximum number of concurrent reconciles for all controllers.")
 	fs.BoolVar(&o.CleanUpOldLogFile, "cleanup-old-cli-logs", true, "Defines if the operator should delete old fdbcli log files.")
 	fs.DurationVar(&o.LogFileMinAge, "log-file-min-age", 5*time.Minute, "Defines the minimum age of fdbcli log files before removing when \"--cleanup-old-cli-logs\" is set.")
+	fs.IntVar(&o.LogFileMaxAge, "log-file-max-age", 28, "Defines the maximum age to retain old operator log file in number of days.")
+	fs.IntVar(&o.LogFileMaxSize, "log-file-max-size", 250, "Defines the maximum size in megabytes of the operator log file before it gets rotated.")
+	fs.IntVar(&o.MaxNumberOfOldLogFiles, "max-old-log-files", 3, "Defines the maximum number of old operator log files to retain.")
+	fs.BoolVar(&o.CompressOldFiles, "compress", false, "It determines if the rotated log files should be compressed using gzip or not.")
```
Comment (johscheuer) [inline setup/setup.go:83]:
```suggestion
	fs.BoolVar(&o.CompressOldFiles, "compress", false, "Defines if the rotated log files should be compressed using gzip or not.")
```

Just for consistency

**Review by johscheuer — COMMENTED** [review]
One thing otherwise LGTM 👍 Did you tested the changes locally?

File: setup/setup.go (hunk 2 above)
Comment (09harsh) [reply setup/setup.go]: Yes I tested the changes locally

**Review by johscheuer — APPROVED** [review]
LGTM 👍

**Pull request merged**